ndarray/src/main/java/org/tensorflow/ndarray/NdArrays.java

```diff
@@ -441,14 +581,14 @@ public static BooleanNdArray wrap(Shape shape, BooleanDataBuffer buffer) {
     if (value == null) {
       throw new IllegalArgumentException();
     }
-    return ofObjects((Class<T>)value.getClass(), Shape.scalar()).setObject(value);
+    return ofObjects((Class<T>) value.getClass(), Shape.scalar()).setObject(value);
```

Comment:
You probably want a `sparseOfObjects` as well? That will be particularly useful to support sparse tensors of Strings in TF.

Reply:
In numeric sparse objects, the default value is defined as zero or false.  What value should I use for Objects? Null?

Reply:
That's a very good question... in case of Strings, I would say an empty string could be a better pick? Would it make sense to let the caller the ability to provide the default value? 

And can we check what are the default value for sparse tensors of strings in Python?

Reply:
TF Python treats Strings with the `string` data type, while  using the empty String as the default value.. Other datatypes beyond what is supported by `tf.dtypes.DType()` are not allowed. In our case, all we need to do, for now, is add support for `java.lang.String`. Also, the method `tf.sparse.to_dense ` has an optional argument for the default value, so maybe that should be added to the sparse classes, too. Default would be zero, false, or empty string.

Reply:
I will write a String sparse array, and will add in the `defaultValue` setting as an option to the CTOR.

Reply:
I'll suggest to leave it generic, like the dense one, taking a default value in parameter of `sparseOfObjects`. But when we'll deal with tensors, we'll have `TString` which will inherit from it and take care of passing automatically that parameter for the user.

I prefer that to preserve the symmetry between the dense and the sparse arrays in the NdArray library, where a dense array can contain any object and not only strings. Wdyt?

Reply:
I just pushed a new class in sparse, `org.tensorflow.ndarray.impl.sparse.SparseNdArray` that can be used for Objects.
I also wrote a test class, `org.tensorflow.ndarray.impl.sparse.StringSparseNdArrayTest `that shows how to use it.
I also added a `defaultValue`  attribute to each sparse class to specify what the default value is instead of forcing it to always be  zero. Default values, zero, false, and null are available in the CTORs for the respective spares classes.

I also had to change `AbstractNdArray` to sue `Objects.equals() `to handle null values.